### PR TITLE
[DependencyInjection] Add section about Service Closures

### PR DESCRIPTION
**Pull request by HypeMC**
Documents service closures added in https://github.com/symfony/symfony/pull/21770 and https://github.com/symfony/symfony/pull/41176.

### CODE REVIEW

File: service_container/service_closures.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    namespace App\Service;
+
+    use Symfony\Component\Mailer\MailerInterface;
+
+    class MyService
+    {
+        /**
+         * @var \Closure
+         */
+        private $mailer;
```
Comment (HeahDude) [inline service_container/service_closures.rst]:
```suggestion
        /**
         * @var callable(): MailerInterface
         */
        private \Closure $mailer;
```
?

File: service_container/service_closures.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        use App\Service\MyService;
+        use Symfony\Component\DependencyInjection\Argument\ServiceClosureArgument;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        return function (ContainerConfigurator $configurator) {
+            $services = $configurator->services();
+
+            $services->set(MyService::class)
+                ->args([new ServiceClosureArgument(new Reference('mailer'))]);
+        };
```
Comment (HeahDude) [inline service_container/service_closures.rst:86]: What about adding a `seealso` directive here pointing to service subscribers to be consistent?

File: service_container/service_closures.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        use App\Service\MyService;
+        use Symfony\Component\DependencyInjection\Argument\ServiceClosureArgument;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        return function (ContainerConfigurator $configurator) {
+            $services = $configurator->services();
+
+            $services->set(MyService::class)
+                ->args([new ServiceClosureArgument(new Reference('mailer'))]);
```
Comment (HeahDude) [inline service_container/service_closures.rst:85]: Note to mergers: this should be updated to:
```suggestion
                ->args([service_closure('mailer')]);
```
when merged up in 5.4.

**Review by HeahDude — APPROVED** [review]
Thank you for documenting this!

File: service_container/service_closures.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
+.. index::
+    single: DependencyInjection; Service Closures
+
+Service Closures
+================
+
```
Comment (OskarStark) [inline service_container/service_closures.rst:6]:
```suggestion

.. versionadded:: 4.4

    Service clousures were introduced in Symfony 4.4.

```

Comment (HeahDude) [reply service_container/service_closures.rst:6]: Actually it was 3.3 :).

Comment (OskarStark) [reply service_container/service_closures.rst:6]: argh, ok lets ignore it then 😄

File: service_container/service_closures.rst (hunk 3 above)
Comment (HypeMC) [reply service_container/service_closures.rst:85]: There's a separate PR for that https://github.com/symfony/symfony-docs/pull/15731

File: service_container/service_closures.rst (hunk 1 above)
Comment (HypeMC) [reply service_container/service_closures.rst]: Done.

File: service_container/service_closures.rst (hunk 2 above)
Comment (HypeMC) [reply service_container/service_closures.rst:86]: Good idea, done.

**Comment by OskarStark** [comment]
As 4.4 is not maintained anymore we should only work on 5.4, can you merge the two PRs?

**Comment by HypeMC** [comment]
@OskarStark Done, #15731 should be good to go.

**Pull request closed without merging**